airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_incremental.py

```diff
@@ -170,70 +170,92 @@ async def test_read_sequential_slices(
 
-        mutating_stream_name_to_per_stream_state = dict()
-        for idx, state_message in enumerate(unique_state_messages):
-            assert state_message.type == Type.STATE
+            states_1 = filter_output(output_1, type_=Type.STATE)
+            # We sometimes have duplicate identical state messages in a stream which we can filter out to speed things up
+            unique_state_messages = [message for index, message in enumerate(states_1) if message not in states_1[:index]]
+            if checkpoint_testing_strategy == CheckpointingStrategies.use_latest_state:
+                unique_state_messages = unique_state_messages[-1:]
 
-            # if first state message, skip
-            # this is because we cannot assert if the first state message will result in new records
-            # as in this case it is possible for a connector to return an empty state message when it first starts.
-            # e.g. if the connector decides it wants to let the caller know that it has started with an empty state.
-            if idx == 0:
-                continue
+            # Important!
 
-            # if last state message, skip
-            # this is because we cannot assert if the last state message will result in new records
-            # as in this case it is possible for a connector to return a previous state message.
-            # e.g. if the connector is using pagination and the last page is only partially full
-            if idx == len(unique_state_messages) - 1:
-                continue
+            # There is only a small subset of assertions we can make
+            # in the absense of enforcing that all connectors return 3 or more state messages
+            # during the first read.
 
-            # if batching required, and not a sample, skip
-            if len(unique_state_messages) >= min_batches_to_test and idx % sample_rate != 0:
+            # To learn more: https://github.com/airbytehq/airbyte/issues/29926
+            if len(unique_state_messages) == 0:
                 continue
 
-            state_input, mutating_stream_name_to_per_stream_state = self.get_next_state_input(
-                state_message, mutating_stream_name_to_per_stream_state, is_per_stream
-            )
-
-            output_N = await docker_runner.call_read_with_state(connector_config, configured_catalog_for_incremental, state=state_input)
-            records_N = filter_output(output_N, type_=Type.RECORD)
-            assert (
-                records_N
-            ), f"Read {idx + 2} of {len(unique_state_messages)} should produce at least one record.\n\n state: {state_input} \n\n records_{idx + 2}: {records_N}"
+            if checkpoint_testing_strategy == CheckpointingStrategies.use_progression:
+                if len(unique_state_messages) < 3:
+                    continue
+
+            # For legacy state format, the final state message contains the final state of all streams. For per-stream state format,
```

Comment:
can we remove support for legacy state format? It's not supported by the CDK anymore

Reply:
I wonder if the problem highlighted by this PR started to happen when we actually started to use per stream state message.

Reply:
**can we remove support for legacy state format? It's not supported by the CDK anymore** - yes we can. I will remove it

Reply:
@alafanechere We have encountered such problems periodically before, but I can't specify what was the reason